### PR TITLE
[AGENT-3741] Download Agent JAR from public repository

### PR DESCRIPTION
**Pull request by amarmudrankit**
In the attempt to remove the agent jar from MLOps python wheel we modify how DRUM gets the agent JAR and export its path for its embedded use case

# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

### CODE REVIEW

File: docker/dropin_env_base/Dockerfile [hunk 1]
```diff
@@ -10,7 +12,9 @@ RUN apt update && apt upgrade -y && apt install -y python3 python3-pip python3-d
 
 COPY requirements.txt requirements.txt
 COPY datarobot-mlops-*.jar /opt/jars/
+COPY mlops-agent-*.jar /opt/jars/
 
 ENV DRUM_JAVA_SHARED_JARS=/opt/jars/*
+ENV MLOPS_MONITORING_AGENT_JAR_PATH=/opt/jars/mlops-agent-${DATAROBOT_MLOPS_VERSION}.jar
```
Comment (amarmudrankit) [inline docker/dropin_env_base/Dockerfile:18]: This is the environment variable that we use to support the embedded agent case.

File: docker/dropin_env_base/build_image.sh [hunk 2]
```diff
@@ -15,4 +15,4 @@ pwd
 
 echo "Building docker image: ${IMAGE_NAME}:${IMAGE_TAG}"
 DATAROBOT_MLOPS_VERSION=8.2.7 ${SCRIPT_DIR}/pull_artifacts.sh
-docker build -t ${IMAGE_NAME}:${IMAGE_TAG} .
+docker build --build-arg DATAROBOT_MLOPS_VERSION=8.2.7 -t ${IMAGE_NAME}:${IMAGE_TAG} .
```
Comment (yakov-g) [inline docker/dropin_env_base/build_image.sh:18]: maybe:
```
export DATAROBOT_MLOPS_VERSION=8.2.7
${SCRIPT_DIR}/pull_artifacts.sh
docker build --build-arg DATAROBOT_MLOPS_VERSION=${DATAROBOT_MLOPS_VERSION} -t ${IMAGE_NAME}:${IMAGE_TAG} .
```

**Comment by yakov-g** [comment]
jenkins run test_integration_r_environment test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_drop_in_envs test_inference_model_templates test_training_model_templates

**Review by yakov-g — APPROVED** [review]

**Review by DrewParallel — APPROVED** [review]

**Pull request merged**